### ISSUE TITLE
Client ID 11 - Coordinate editing mode in GFI

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/austrocontrol-C125#234]
(d) poor/incomplete display of coordinates and buttons when on small display (tested in iPhone 7, Safari and chrome + iPad Safari and  Chrome)

screenshots attached

[https://app.zenhub.com/files/111421835/0444c4b8-f3e8-4bda-8708-05643e248546/download](https://app.zenhub.com/files/111421835/0444c4b8-f3e8-4bda-8708-05643e248546/download)

**Comment by tdipisa** [comment]
> (d) poor/incomplete display of coordinates and buttons when on small display (tested in iPhone 7, Safari and chrome + iPad Safari and Chrome)

Effectively there are layout problems in some cases. We have to check this.

**Comment by tdipisa** [comment]
Dear @michegg, I've check it better by looking at the MS instance [here](https://austrocontrol.geo-solutions.it/austrocontrol-ms2/). See points below please.

**1.**
You have attached a screenshot with the side panel open. It is strange because for mobile devices it shouldn't be there: a panel should open at the bottom of the screen.

![2020_12_01_12_28_23_coordinate_display_ipad_2_ png_Foto](https://user-images.githubusercontent.com/1280027/100735121-f9143f80-33d0-11eb-817c-1e9632226b2b.png)

Indeed, this is what I have for example for **iPad**  using the browser developer tool in Chrome: and all seems to be fine there.

![2020_12_01_11_57_20_AustroControl_Custom_MapStore2_Project](https://user-images.githubusercontent.com/1280027/100732107-95881300-33cc-11eb-9550-6cbc86802fce.png)

Which is the exact device you have used to test this? Did you use some emulator?

**2.**
I've tested it also using my iPhone 11 and this is the result (the same of your screenshots): some fixes to decimal coordinates fields and some button vertical alignement for areonautical coordinates are needed I think

![2020_12_01_12_34_07_Foto](https://user-images.githubusercontent.com/1280027/100735620-d2a2d400-33d1-11eb-84dd-070073c69196.png)

**3.**
It is strange that you have the measurement panel available in iPad.

![image](https://user-images.githubusercontent.com/1280027/100737198-244c5e00-33d4-11eb-978d-98869b38a262.png)

It isn't set to be available by default for mobile devices, below it is what I have for iPad using the browser developer tool in Chrome:

![image](https://user-images.githubusercontent.com/1280027/100737304-480fa400-33d4-11eb-83f3-f1dc174f0ec5.png)

Therefore, I'm wondering how did you test it and with which device. Can you provide some feedback for this?

**Comment by michegg** [comment]
@tdipisa I tested on an iPad pro 12,9'' (no emulator) with Google Chrome and still get the same results as reported.
Using the chrome developer tools (set to iPad Pro), functions are also not restricted to the mobile version, but the layout is working fine.

So maybe it's more a problem of small desktop devices?

**Comment by tdipisa** [comment]
Dear @michegg 

> @tdipisa I tested on an iPad pro 12,9'' (no emulator) with Google Chrome and still get the same results as reported.

Ok.

> Using the chrome developer tools (set to iPad Pro), functions are also not restricted to the mobile version,

When you use the Chrome developer tool to switch to mobile mode, you have to refresh the page to effectively keep this mode and have the UI loaded according to the mobile type you have set. If you don't refresh the page the browser mode still remain the previous one (with just a smaller screen than for desktop). If you do that, you can see that certain tools are not available in mobile mode, according to the default MapStore configuration in localConfig.

>but the layout is working fine.

Yes, indeed.

> So maybe it's more a problem of small desktop devices?

In MapStore we identify the device type by checking the userAgent header: this is a first check we do also to apply a different code if/where needed. Then to manage cross device compatibility we are also using general **flex** rules in the CSS that are automatically managed by the browser to properly adapt the UI to the device itself. This is the general procedure we can adopt since there are a lot of existing mobile devices: specific mobile devices will require the introduction of dedicated rules, that is something that we can't do now unfortunately.

The only way we have at the moment to check this cross device compatibility is using some emulator, and our developers usually use the Chrome developer tool for this since, otherwise, we should have all physical devices available, and this isn't our case unfortunately. 

Using the Chrome developer tool, we are not able to replicate the issue that I've mentioned before, that is the only one I found using my Iphone 11 (not available to developers):

![2020_12_01_12_34_07_Foto](https://user-images.githubusercontent.com/1280027/100735620-d2a2d400-33d1-11eb-84dd-070073c69196.png)

In short, I don't think we can do something more for this now. I'm sorry.

**Comment by tdipisa** [comment]
@michegg do you have some feedback for this?

**Comment by michegg** [comment]
@tdipisa: sure, I understand this. I got the chance to test the map on iPad Pro privately as of course it is really useful for pilots to bring such devices into the cockpit. Nevertheless it shouldn't be a problem to use the desktop version on such a screen size.

It seems to be an iOS problem / or that Chrome doesn't emulate those devices correctly.
- The problem you found on your iPhone 11 also isn't correctly displayed in Chrome dev. tools. While the way it is displayed, also isn't desirable --> see screenshot. Input fields should be one above the other, in order to be usable.
![emulate_iphone](https://user-images.githubusercontent.com/57953447/101621337-e7a4e600-3a15-11eb-8a44-13d203a950d0.PNG)

- Also the mobile version on iPad doesn't display the fields correctly. When I request the mobile version on the iPad I get the same results as you do on your iPhone
![ipad_mobile](https://user-images.githubusercontent.com/57953447/101621241-c9d78100-3a15-11eb-899d-70947238d564.jpg)


- While when I request the desktop version on a Samsung Tab S2 the coordinate input fields are displayed correctly.

So maybe it is desirable for you to at least test on one iOS device to keep MapStore useable on Apple devices.

**Comment by tdipisa** [comment]
> The problem you found on your iPhone 11 also isn't correctly displayed in Chrome dev. tools. While the way it is displayed, also isn't desirable --> see screenshot. Input fields should be one above the other, in order to be usable.

> Also the mobile version on iPad doesn't display the fields correctly. When I request the mobile version on the iPad I get the same results as you do on your iPhone

Yes, you get the point: Chrome doesn't emulate Apple devices properly but, unfortunately, this the only way for our developers to check this things at the moment. Therefore, what I can do now is to open an issue in MS for a future improvement for iOS devices.

**Comment by FlorianBruckner** [comment]
This looks like a problem related to recent Safari. I can reproduce on both the desktop version of Safari and in the Emulator:
![image](https://user-images.githubusercontent.com/2104853/104421596-d8b4d480-557b-11eb-8755-84d845e4f301.png)
![image](https://user-images.githubusercontent.com/2104853/104421667-f5e9a300-557b-11eb-8527-197c3d0b9609.png)

I guess this has to do with the flex layout used in the markup

**Comment by FlorianBruckner** [comment]
It is not present in the open source demo version.

![image](https://user-images.githubusercontent.com/2104853/104421924-5973d080-557c-11eb-8e8d-44ec6509e900.png)

But there is a markup difference. You can see the coordinate editor highlighted in the inspector panel

**Comment by FlorianBruckner** [comment]
![image](https://user-images.githubusercontent.com/2104853/104422115-9770f480-557c-11eb-9d92-8a8953f14417.png)

Manipulating the DOM in the emulator to wrap the second input into a div with classes coordinate and lon fixes the problem:

![image](https://user-images.githubusercontent.com/2104853/104422381-fc2c4f00-557c-11eb-9598-92898ebd1c6a.png)

**Comment by FlorianBruckner** [comment]
It looks like the root cause of this is a problem that Safari has with "display: table" children in a "display: flex" container. The Input Group is a "display: table" in Bootstrap 3.

There are multiple ways around this - wrapping the input-group in a div seems to be the easiest way.
